.github/workflows/format.yml

```diff
@@ -0,0 +1,56 @@
+name: Format check
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**.py'
+
+jobs:
+  generate:
+    runs-on: ubuntu-latest
+    outputs:
+      files: ${{ steps.git-diff-filter.outputs.files }}
+    name: Detect added and changed files
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: GrantBirki/git-diff-action@v2
+        id: git-diff-action
+        with:
+          base_branch: origin/main
+          search_path: '**.py'
+          # Ignore deleted files
+          git_options: '--no-color --diff-filter=d'
+
+      - name: Filter json diff
```

Comment:
Filter what? It should be possible to have a clue without reading the code.

Reply:
Doesn't `file_output_only` do this? It's "highly recommended"

Reply:
Used the recommended json file output.